spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigUtils.java

```diff
@@ -227,13 +224,34 @@ public static MultipleSourcesContainer processNamedData(List<StrippedSourceConta
 		return new MultipleSourcesContainer(foundSourceNames, data);
 	}
 
+	static boolean processSource(boolean includeDefaultProfileData, Environment environment, String sourceName,
+			Map<String, String> sourceRawData) {
+		Set<String> activeProfiles = Arrays.stream(environment.getActiveProfiles()).collect(Collectors.toSet());
```

Comment:
Can we use a List here?  The order of profiles is generally important so I want to make sure we maintain that order